tensorhive/controllers/restriction/update_restriction_controller.py

```diff
@@ -0,0 +1,41 @@
+from tensorhive.authorization import admin_required
+from tensorhive.models.Restriction import Restriction
+from sqlalchemy.orm.exc import NoResultFound
+from tensorhive.config import API
+import logging
+log = logging.getLogger(__name__)
+R = API.RESPONSES['restriction']
+G = API.RESPONSES['general']
+
+
+@admin_required
+def update(id, newValues):
+    new_values = newValues
+    allowed_fields = {'name', 'start', 'end', 'isGlobal'}
+    try:
+        assert set(new_values.keys()).issubset(allowed_fields), 'invalid field is present'
+        restriction = Restriction.get(id)
+
+        for field_name, new_value in new_values.items():
+            # Mapping API field name to column used by Restriction model
+            if field_name in ['start', 'end']:
+                # start -> starts_at, end -> ends_at
```

Comment:
hackyyy... how about creating a method like this:

```python3
def to_db_column():
    return {
        'name': 'name',
        'start': 'starts_at',
        'end': 'ends_at',
        'isGlobal': 'is_global'
    }
```

Then you could use it like this: `field_name = to_db_column()[field_name]`